ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoFilteredTransactionBuilder.kt

```diff
@@ -0,0 +1,68 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.ledger.utxo.Command
+import net.corda.v5.ledger.utxo.ContractState
+import net.corda.v5.ledger.utxo.StateRef
+import java.security.PublicKey
+import java.util.function.Predicate
+
+interface UtxoFilteredTransactionBuilder {
+
+    // and time window
+    // no benefit of predicates for inputs
```

Comment:
I'm not following this comment. Is this suggesting that notary information implies time window information too?

If so, this doesn't feel right to me - that would be an implementation detail and it's not clear at all that would really be the case from the interface. Plus, an arbitrary filtered transaction might want to send a tear-off to another node with time window but not notary?

Logically notary and time windows are different things.

Reply:
@blsemo @mnesbit I can implement this, but I think this disagrees with what @mnesbit said to me the other day.

Reply:
If it's not a lot of effort to have a separate `withNotary` and `withTimeWindow` function, that might be worthwhile - we treat them as independent entities on the API.

Reply:
Done.